homeassistant/auth/__init__.py

```diff
@@ -257,7 +263,44 @@ def get_auth_mfa_module(self, module_id: str) \
         if not user.system_generated and client_id is None:
             raise ValueError('Client is required to generate a refresh token.')
 
-        return await self._store.async_create_refresh_token(user, client_id)
+        if token_type is None:
+            if user.system_generated:
+                token_type = models.TOKEN_TYPE_SYSTEM
+            else:
+                token_type = models.TOKEN_TYPE_NORMAL
+
+        if user.system_generated and token_type != models.TOKEN_TYPE_SYSTEM:
+            raise ValueError(
+                'System generated users can only have system type '
+                'refresh tokens')
+
+        if not user.system_generated and \
+                token_type == models.TOKEN_TYPE_SYSTEM:
+            raise ValueError(
+                'Only system generated users can have system type '
+                'refresh tokens')
+
+        if (token_type == models.TOKEN_TYPE_LONG_LIVED_ACCESS_TOKEN and
+                (client_id is None or client_name is None)):
```

Comment:
`client_id` should be `None`. Client ID is what is used when a user generates a token via an external client. When creating a long lived token, they use the UI and no external client is involved.